app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -86,4 +87,21 @@ def registration_url
     msg = N_('URL in :url parameter is missing a scheme, please set http:// or https://')
     fail Foreman::Exception.new(msg)
   end
+
+  def global_setup_insights(organization, location, host_group, operatingsystem)
+    return if params['setup_insights'].to_s.blank?
+
+    from_host = Host.new(organization: organization, location: location, hostgroup: host_group, operatingsystem: operatingsystem).params['host_registration_insights']
```

Comment:
ideally the host object should be built outside of this method, so you can later reuse it for setting up rex parameter, but it's ok to keep that to the rex parameter PR too